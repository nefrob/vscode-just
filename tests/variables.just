
# Aliases

alias b := build

build:
  echo 'Building!'

# Settings

set allow-duplicate-recipes := false

# Variables and substitution

tmpdir  := `mktemp -d`
version := "0.2.7"
tardir  := tmpdir / "awesomesauce-" + version
tarball := tardir + ".tar.gz"

foo := "a/"
bar := foo / "b"

export RUST_BACKTRACE := "1"

