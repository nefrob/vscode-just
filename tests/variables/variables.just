
# Aliases

alias b := build

build:
  echo 'Building!'

# Variables and substitution

tmpdir  := `mktemp -d`
version := "0.2.7"
tardir  := tmpdir / "awesomesauce-" + version
tarball := tardir + ".tar.gz"

foo := "a/"
bar := foo / "b"

export RUST_BACKTRACE := "1"

