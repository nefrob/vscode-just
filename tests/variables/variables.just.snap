>
># Aliases
#^^^^^^^^^ source.just comment.line.number-sign.just
>
>alias b := build
#^^^^^ source.just keyword.other.reserved.just
#     ^ source.just
#      ^ source.just variable.name.alias.just
#       ^ source.just
#        ^^ source.just keyword.operator.assignment.just
#          ^ source.just
#           ^^^^^ source.just variable.other.just
>alias help-123 := some-recipe
#^^^^^ source.just keyword.other.reserved.just
#     ^ source.just
#      ^^^^^^^^ source.just variable.name.alias.just
#              ^ source.just
#               ^^ source.just keyword.operator.assignment.just
#                 ^ source.just
#                  ^^^^^^^^^^^ source.just variable.other.just
>
># Variables and substitution
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.just comment.line.number-sign.just
>
>tmpdir  := `mktemp -d`
#^^^^^^ source.just variable.other.just
#      ^^ source.just
#        ^^ source.just keyword.operator.assignment.just
#          ^ source.just
#           ^ source.just string.interpolated.just
#            ^^^^^^^^^ source.just source.shell
#                     ^ source.just string.interpolated.just
>version := "0.2.7"
#^^^^^^^ source.just variable.other.just
#       ^ source.just
#        ^^ source.just keyword.operator.assignment.just
#          ^ source.just
#           ^ source.just string.quoted.double.just string.quoted.double.just
#            ^^^^^ source.just string.quoted.double.just
#                 ^ source.just string.quoted.double.just
>tardir  := tmpdir / "awesomesauce-" + version
#^^^^^^ source.just variable.other.just
#      ^^ source.just
#        ^^ source.just keyword.operator.assignment.just
#          ^^^^^^^^ source.just
#                  ^ source.just keyword.operator.path-join.just
#                   ^ source.just
#                    ^ source.just string.quoted.double.just string.quoted.double.just
#                     ^^^^^^^^^^^^^ source.just string.quoted.double.just
#                                  ^ source.just string.quoted.double.just
#                                   ^ source.just
#                                    ^ source.just keyword.operator.concat.just
#                                     ^^^^^^^^ source.just
>tarball := tardir + ".tar.gz"
#^^^^^^^ source.just variable.other.just
#       ^ source.just
#        ^^ source.just keyword.operator.assignment.just
#          ^^^^^^^^ source.just
#                  ^ source.just keyword.operator.concat.just
#                   ^ source.just
#                    ^ source.just string.quoted.double.just string.quoted.double.just
#                     ^^^^^^^ source.just string.quoted.double.just
#                            ^ source.just string.quoted.double.just
>
>foo := "a/"
#^^^ source.just variable.other.just
#   ^ source.just
#    ^^ source.just keyword.operator.assignment.just
#      ^ source.just
#       ^ source.just string.quoted.double.just string.quoted.double.just
#        ^^ source.just string.quoted.double.just
#          ^ source.just string.quoted.double.just
>bar := foo / "b"
#^^^ source.just variable.other.just
#   ^ source.just
#    ^^ source.just keyword.operator.assignment.just
#      ^^^^^ source.just
#           ^ source.just keyword.operator.path-join.just
#            ^ source.just
#             ^ source.just string.quoted.double.just string.quoted.double.just
#              ^ source.just string.quoted.double.just
#               ^ source.just string.quoted.double.just
>
># Exports
#^^^^^^^^^ source.just comment.line.number-sign.just
>
>export RUST_BACKTRACE := "1"
#^^^^^^ source.just keyword.other.reserved.just
#      ^ source.just
#       ^^^^^^^^^^^^^^ source.just variable.other.just
#                     ^ source.just
#                      ^^ source.just keyword.operator.assignment.just
#                        ^ source.just
#                         ^ source.just string.quoted.double.just string.quoted.double.just
#                          ^ source.just string.quoted.double.just
#                           ^ source.just string.quoted.double.just
>
>